docs/ref/contrib/gis/gdal.txt

```diff
@@ -1731,6 +1770,55 @@ Key              Default                           Usage
 ``offset``       ``(0, 0)``                        Passed to the :meth:`~GDALBand.data` method
 ================ ================================= ======================================================
 
+.. _gdal-raster-vsimem:
+
+Using GDAL's Virtual Filesystem
+-------------------------------
+GDAL has an internal memory based filesystem, which allows treating blocks of
+memory as files. It can be used to read and write :class:`GDALRaster` objects
+from and to binary file buffers.
+
+This is useful in web contexts, where rasters might be obtained as a buffer
+from a remote storage, or to returned from a view as a file without writing
+it to disk first.
+
+:class:`GDALRaster` objects are created in the virtual filesystem when a
+``bytes`` object is provided as input, or when the file path starts with
+``/vsimem/``.
+
+Input provided as ``bytes`` has to be a full binary representation of a file.
+For instance::
+
+    # Read a raster as file object from a remote source.
+    >>> import urllib
+    >>> dat = urllib.urlopen('http://example.com/raster.tif').read()
+    # Instantiate a raster from the bytes object.
+    >>> rst = GDALRaster(dat)
+    # The name starts with /vsimem/, indicating that the raster lives in the
+    # virtual filesystem.
+    >>> rst.name
+    /vsimem/da300bdb-129d-49a8-b336-e410a9428dad
+
+To create a new virtual file based raster from scratch, use the ``ds_input``
+dictionary representation, and provide a ``name`` argument that starts with
+``/vsimem/`` (for detail on the dictionary representation see
+:ref:`gdal-raster-ds-input`). For virtual file based rasters, the
+:attr:`~GDALRaster.vsi_buffer` will return the ``bytes`` representation of the
+raster. 
+::
+
+    # Create a new GTiff raster in a view and return it as a file.
+    >>> rst = GDALRaster({
+    ...     "name": "/vsimem/temporarymemfile",
+    ...     "driver": "tif",
+    ...     "width": 6, "height": 6, "srid": 3086,
+    ...     "origin": [500000, 400000],
+    ...     "scale": [100, -100],
+    ...     "bands": [{"data": range(36), "nodata_value": 99}]
+    ... })
+    >>> return HttpResponse(rast.vsi_buffer, 'image/tiff')
```

Comment:
A return statement doesn't make sense in a shell session. Also, HttpResponse import is missing.